cmd/security-secrets-setup/main.go

```diff
@@ -69,24 +75,52 @@ func main() {
 		return
 	}
 
-	if configFile == "" {
-		// run with other options for pki-init
-		statusCode, err := dispatcherInstance.run()
-		if err != nil {
-			setup.LoggingClient.Error(err.Error())
-		}
+	if checkIfMultipleSubcommands() {
+		setup.LoggingClient.Error("cannot use multiple subcommands, use one at a time")
+		exitInstance.exit(1)
+		return
+	}
+
+	subcmdName := flag.Args()[0]
 
-		exitInstance.exit(statusCode)
+	subcmd, found := subcommands[subcmdName]
+	if !found {
+		setup.LoggingClient.Error(fmt.Sprintf("unsupported subcommand %s", subcmdName))
+		exitInstance.exit(1)
 		return
 	}
 
-	if err := option.GenTLSAssets(configFile); err != nil {
-		setup.LoggingClient.Error(err.Error())
+	if err := subcmd.Parse(flag.Args()[1:]); err != nil {
+		setup.LoggingClient.Error(fmt.Sprintf("error parsing subcommand %s: %v", subcmdName, err))
 		exitInstance.exit(2)
 		return
 	}
 
+	var exitStatusCode int
+	var err error
+
+	switch subcmdName {
+	case "legacy":
+		if err = option.GenTLSAssets(configFile); err != nil {
```

Comment:
we should also confirm that there are no additional arguments to the legacy command here like we do below for generate cache, import, etc.

Reply:
i thought that is checked or handled by subcommand's parse api itself already like invalid flag provided error message will be given in flag's parse as well

Reply:
no it is not, what will happen is that the flagset will just have those arguments defined, there's no way of telling the stdlib flag library "this flagset doesn't take any arguments"

Reply:
ok, but this "legacy" subcommand does accept argument -config flag, though, see the leagcy subcommand flagset definition in the code.

Reply:
the argument `-config` does not show up in `flagset.Args()` because it is parsed as the option, because we are using a flag set and defining the option in the flagset

Reply:
so here is the output of my execution without that checking additional argument logic.  It looks  like the subcomand parse(os.args[1:]) already does the checking for you, doesn't it?  
```
./security-secrets-setup legacy -config ./res/pkisetup-vault.json -extra invalid
level=ERROR ts=2019-09-19T17:25:26.178019174Z app=edgex-security-secrets-setup source=logger.go:105 msg="logTarget cannot be blank, using stdout only"
flag provided but not defined: -extra
Usage of legacy:
  -c string
    	specify JSON configuration file: /path/to/file.json
  -config string
    	specify JSON configuration file: /path/to/file.json
```

Reply:
ok, I think I got what you meant now. I also added some extra tests for these cases.